vowpalwabbitslim/opts.cc

```diff
@@ -0,0 +1,78 @@
+#include "opts.h"
+
+#include <cstdlib>
+#include <cctype>
+
+namespace vw_slim {
+	// && used to avoid constant string copying
+	void find_opt(std::string const& command_line_args, std::string arg_name, std::vector<std::string>& out_values)
+	{
+		// append space to search for '--quadratic '
+		arg_name += ' ';
+
+		// support -q ab -v -q cd
+		for (size_t start = 0; start < command_line_args.size();)
+		{
+			auto idx = command_line_args.find(arg_name, start);
+			if (idx == std::string::npos)
+				return; // no more occurences found, exit
+
+			auto idx_after_arg = idx + arg_name.size();
+
+			// skip all white space
+			for (; idx_after_arg < command_line_args.size() && std::isspace(command_line_args[idx_after_arg]); ++idx_after_arg);
+
+			if (idx_after_arg == command_line_args.size())
+				return;
+
+			if (command_line_args[idx_after_arg] == '-' &&
+				// make sure we allow -5.2 (negative numbers) 
+				!(idx_after_arg + 1 < command_line_args.size() && (command_line_args[idx_after_arg + 1] >= '0' && command_line_args[idx_after_arg + 1] <= '9')))
+			{
+				start = idx_after_arg;
+				continue; // next option found
+			}
+
+			// find next non-white space character
+			auto idx_after_value = idx_after_arg;
```

Comment:
Is there a need to handle single or double quoted values?

Reply:
I haven't encountered such options. Any example?

---
In reply to: [188485791](https://github.com/JohnLangford/vowpal_wabbit/pull/1481#discussion_r188485791) [](ancestors = 188485791)